src/trio/_core/_traps.py

```diff
@@ -45,30 +44,31 @@ class PermanentlyDetachCoroutineObject:
     type[CancelShieldedCheckpoint],
     WaitTaskRescheduled,
     PermanentlyDetachCoroutineObject,
-    object,
+    object,  # For reattach_detached_coroutine_object(), a foreign loop's value.
 ]
 
 
-# Helper for the bottommost 'yield'. You can't use 'yield' inside an async
-# function, but you can inside a generator, and if you decorate your generator
-# with @types.coroutine, then it's even awaitable. However, it's still not a
-# real async function: in particular, it isn't recognized by
-# inspect.iscoroutinefunction, and it doesn't trigger the unawaited coroutine
-# tracking machinery. Since our traps are public APIs, we make them real async
-# functions, and then this helper takes care of the actual yield:
-@types.coroutine
-def _real_async_yield(
-    obj: MessageType,
-) -> Generator[MessageType, None, None]:
-    return (yield obj)
+class _AsyncYield:
+    """Helper for the bottommost 'yield'.
 
+    You can't use 'yield' inside an async function, so implement an awaitable object to do so.
```

Comment:
You can use `yield` inside an async function! It just doesn't do what we want.

Reply:
Quite true.